File: i2f-springboot/src/main/java/i2f/springboot/secure/vue/secure/base64-obfuscator.js
import Random from "./random";

const Base64Obfuscator = {
  OBF_PREFIX() {
    return "$.";
  },
  encode(bs4, prefix) {
    let builder = "";
    if (prefix) {
      builder += this.OBF_PREFIX();
    }
    for (let i = 0; i < bs4.length; i++) {
      if (i % 2 == 0 && bs4.charAt(i) != '=') {
        let ird = Random.nextLowerInt(16);
        if (ird < 10) {
          builder += (String.fromCharCode('0'.charCodeAt(0) + ird));
        } else {
          builder += (String.fromCharCode('A'.charCodeAt(0) + (ird - 10)));
        }
      }
      builder += (bs4.charAt(i));
    }
    return builder;
  },
  decode(sob) {
    let str = sob;
    if (sob.startsWith(this.OBF_PREFIX())) {
      str = sob.substring(this.OBF_PREFIX().length);
    }
    let builder = '';
    let i = 0;
    while (i < str.length) {
      if (i % 3 == 0 && str.charAt(i) != '=') {
        i++;
        continue;
      }
      builder += (str.charAt(i));
      i++;
    }
    return builder;
  }
};


export default Base64Obfuscator;

File: i2f-core/src/main/java/i2f/core/digest/Base64Obfuscator.java
package i2f.core.digest;

import i2f.core.check.CheckUtil;

import java.util.Random;

/**
 * @author ltb
 * @date 2022/5/6 15:27
 * @desc base64 混淆器实现
 * 每隔原始的base64串2位，随机插入一个hex的char
 */
public class Base64Obfuscator {
    public static final String OBF_PREFIX ="$.";
    public static String encode(String bs4,boolean prefix) {
        if (CheckUtil.isEmptyStr(bs4)) {
            return bs4;
        }
        Random random = new Random();
        StringBuilder builder = new StringBuilder();
        if (prefix) {
            builder.append(OBF_PREFIX);
        }
        int jpc = (bs4.length() + random.nextInt(31)) % 10;
        builder.append((char) ('0' + jpc));
        for (int p = 0; p < jpc; p++) {
            int ird = random.nextInt(16);
            if (ird < 10) {
                builder.append((char)('0'+ird));
            }else{
                builder.append((char)('A'+(ird-10)));
            }
        }
        jpc=jpc%2+2;
        for(int i=0;i<bs4.length();i++){
            if(i%jpc==0 && bs4.charAt(i)!='='){
                int ird=random.nextInt(16);
                if(ird<10){
                    builder.append((char)('0'+ird));
                }else{
                    builder.append((char)('A'+(ird-10)));
                }
            }
            builder.append(bs4.charAt(i));
        }
        return builder.toString();
    }
    public static String decode(String sob) {
        if (CheckUtil.isEmptyStr(sob)) {
            return sob;
        }
        String str = sob;
        if (sob.startsWith(OBF_PREFIX)) {
            str = sob.substring(OBF_PREFIX.length());
        }
        StringBuilder builder = new StringBuilder();
        int jpc = str.charAt(0) - '0';
        str = str.substring(1 + jpc);
        jpc = jpc % 2 + 3;
        int i = 0;
        while (i < str.length()) {
            if(i%jpc==0 && str.charAt(i)!='='){
                i++;
                continue;
            }
            builder.append(str.charAt(i));
            i++;
        }
        return builder.toString();
    }

}
